### DISCUSSION TITLE
PSR-3 Logging

### DISCUSSION THREAD

**Discussion by eric-lamb62** [ExpressionEngine/ExpressionEngine#3821] [Feature Requests]
**NOTE THIS IS BASED ON AN ABSTRACT DISCUSSION AT ORLANDO EECONF FROM THE COMMUNITY**

**Is your feature request related to a problem? Please describe.**
At the moment, logging in EE is pretty archaic so an improvement up to modern standards through the PSR-3 implementation would be a dramatic improvement. 

**Describe the solution you'd like**
That we keep the existing interface intact and either change the internals to use a PSR-3 compliant implementation or create a new stand alone layer that can replace the `EE_Log` object iteratively. 

**Describe alternatives you've considered**
None. An improvement is needed and PSR-3 is the gold standard of logging with PHP. 

**Teachability, Documentation, Adoption, Migration Strategy**
Out of the box, nothing would have to be done by the end user but for those devs who are fluent they'll be able to make use of the driver based design to allow logging to other endpoints (database, email, sms, rest, anything a dev can dream up and build for).

- [ x ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@mithra62 I have nothing to say on this front, but we already have similar thread: https://github.com/ExpressionEngine/ExpressionEngine/discussions/2996

Could you post your thoughts there and then we'll just close this one
